rust_snuba/src/processors/replays.rs

```diff
@@ -0,0 +1,593 @@
+use anyhow::Context;
+use chrono::{DateTime, Utc};
+use md5;
+use rust_arroyo::backends::kafka::types::KafkaPayload;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+use uuid::Uuid;
+
+use crate::types::{InsertBatch, KafkaMessageMetadata, RowData};
+
+pub fn process_message(
+    payload: KafkaPayload,
+    metadata: KafkaMessageMetadata,
+) -> anyhow::Result<InsertBatch> {
+    let payload_bytes = payload.payload().context("Expected payload")?;
+
+    let replay_message: ReplayMessage = serde_json::from_slice(payload_bytes)?;
+    let replay_payload: ReplayPayload = serde_json::from_slice(&replay_message.payload)?;
```

Comment:
This type can be inferred.
```suggestion
    let replay_payload = serde_json::from_slice(&replay_message.payload)?;
```